internal/partitions/partitions.go

```diff
@@ -360,17 +384,24 @@ func getOuterIDBound(ctx context.Context, subLogger *logger.Logger, retryer *ret
 	}
 
 	var docID interface{}
+
+	var pipeline []bson.D
+	if len(globalFilter) > 0 {
+		pipeline = append(pipeline, bson.D{{"$match", globalFilter}})
+	}
+	pipeline = append(pipeline, []bson.D{
+		bson.D{{"$sort", bson.D{{"_id", sortDirection}}}},
```

Comment:
Nit: You can omit the outer `bson.D` here since it’s implicit in the slice type.